packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -168,6 +196,9 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
                 0,
                 txCallback.numTxs
             );
+            for (uint j = txIdx; j < txIdx + txCallback.numTxs; j++) {
+                preApprovedTxs[j] = true;
```

Comment:
Is it possible that preApprovedTxs[j] == false? If not, then there is unnecessary to build this preApprovedTxs array, just check auxilaryData.approved == true will be enough.

Reply:
We need to make sure `approved` is only true for transactions processed and verified in the pool contracts. If the conditional transactions isn't checked there the value has to be false so the transaction is still verified by the exchange contract.

Reply:
Okay, that makes sense.